pkgs/development/haskell-modules/configuration-darwin.nix

```
@@ -391,6 +391,10 @@ self: super:
       '';
     }) super.rio;
 
+    # Don't use homebrew icu on macOS
+    # https://github.com/NixOS/nixpkgs/issues/462046
+    text-icu = appendConfigureFlags [ "-f-homebrew" ] super.text-icu;
```

Sorry, forgot to say, this is a little more robust.

```suggestion
    text-icu = disableCabalFlag "homebrew" super.text-icu;
```

---

👍